iFixit/react-commerce
# Add store home page under feature flag

closes #803 

This PR adds a feature flag for the new store home page. If `NEXT_PUBLIC_FLAG__STORE_HOME_PAGE_ENABLED` is set to `true` the new store home page will be visible.
The template implementing the store home page has a more generic name  because it can be reused to build essentially any landing page.

## QA

I have enabled the flag only for `react-commerce`, so these steps are to make sure that the store home is not visible from the production deployments.

1. Visit ["production" Vercel preview](https://react-commerce-prod-5a2pxjjzo-ifixit.vercel.app/Store)
2. Verify that [`/Store`](https://react-commerce-prod-5a2pxjjzo-ifixit.vercel.app/Store) returns a 404
3. Visit ["staging" Vercel preview](https://react-commerce-git-add-store-home-feature-flag-ifixit.vercel.app/Store) (flag enabled)
4. Verify that [`/Store`](https://react-commerce-git-add-store-home-feature-flag-ifixit.vercel.app/Store) renders a dummy page
